packages/ui/src/ui/pages/navigation/Navigation/NavigationError/helpers/helpers.ts

```diff
@@ -0,0 +1,84 @@
+import {getYtErrorCode} from '../../../../../utils/errors';
+import {YTError} from '../../../../../../@types/types';
+import {YTErrors} from '../../../../../rum/constants';
+import {UnipikaValue} from '../../../../../components/Yson/StructuredYson/StructuredYsonTypes';
+
+export type ErrorName = keyof typeof YTErrors;
+
+/**
+ * should be: (typeof YTErrors)[keyof typeof YTErrors]
+ * after migrating javascript-wrapper on TS
+ */
+export type ErrorCode = 500 | 901;
+
+type NoAccessTitlePayload = {
+    username: string;
+    permissions: Array<UnipikaValue>;
+    path: string;
+};
+
+type NoPathTitlePayload = {
+    path: string;
+};
+
+type TitlePayload = NoAccessTitlePayload & NoPathTitlePayload;
+
+type ErrorInfo = {
+    [key in ErrorCode]: {
+        getTitle: (payload: TitlePayload) => string;
+    };
+};
+
+export const ErrorsInfo: ErrorInfo = {
+    901: {
+        getTitle: (payload: NoAccessTitlePayload) => {
+            const {username, permissions, path} = payload;
+            const permission = permissions.map((perm: UnipikaValue) => perm.$value).join(' | ');
+            return `User ${username} does not have "${permission}" access to node "${path}"`;
+        },
+    },
+    500: {
+        getTitle: (payload: NoPathTitlePayload) => {
+            const {path} = payload;
+            return `Path "${path}" does not exist`;
+        },
+    },
+};
+
+export function getErrorTitle(error: YTError, path?: string): string {
+    const {attributes} = error;
+
+    const code = getErrorCode(error);
+
+    if (!code) return 'An unexpected error occurred';
+
+    const title = ErrorsInfo[code].getTitle({
+        path: path || '',
+        username: attributes?.user.$value || '',
+        permissions: attributes?.permission || '',
+    });
+
+    return title;
+}
+
+export function getErrorCode(error: YTError): ErrorCode | undefined {
```

Comment:
It is better to highlight that we return first valid error code, so lets name it like getFirstErrorCode. Or may be you know a better name.